gateway/src/config.rs

```diff
@@ -0,0 +1,535 @@
+//! User configuration for shards.
+
+use crate::{tls::TlsContainer, EventTypeFlags, Session};
+use std::{
+    fmt::{Display, Formatter, Result as FmtResult},
+    sync::Arc,
+};
+use twilight_gateway_queue::{LocalQueue, Queue};
+use twilight_model::gateway::{
+    payload::outgoing::{identify::IdentifyProperties, update_presence::UpdatePresencePayload},
+    Intents,
+};
+
+/// Identifier of a [shard], including the shard's ID and the total number of
+/// shards in use by the bot.
+///
+/// [shard]: crate::Shard
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct ShardId {
+    /// Current ID of the shard, 0-indexed.
+    current: u64,
```

Comment:
`current` sounds like it's changeable to me. `this` is a better term imo.

Reply:
`this` is generally used as an alias for `self`, eg in future implementation pinning.. will think of an alternative